src/signup/AccountInfo.tsx

```diff
@@ -0,0 +1,177 @@
+// 회원가입 페이지 - 아이디, 비밀번호 등 계정 입력 컴포넌트
+import Image from "next/image";
+import React, { useEffect, useState } from "react";
+import { useSetRecoilState } from "recoil";
+import { signupPageState } from "@shared/recoil/signup";
+import Button from "./Button";
+import CheckAuth from "./CheckAuth";
+import Input from "./Input";
+import Instructions from "./Instructions";
+
+interface FormData {
+  [id: string]: { essential: boolean; value: string };
+  password: { essential: boolean; value: string };
+  checkPassword: { essential: boolean; value: string };
+  phone: { essential: boolean; value: string };
+}
+
+const AccountInfo = () => {
+  const setCurrentPage = useSetRecoilState(signupPageState);
+
+  const [isSubmit, setIsSubmit] = useState(false);
+  const [isDoneInput, setIsDoneInput] = useState(false);
+  const [formData, setFormData] = useState<FormData>({
+    id: { essential: true, value: "" },
+    password: { essential: true, value: "" },
+    checkPassword: { essential: true, value: "" },
+    phone: { essential: true, value: "" },
+  });
+
+  //필수 데이터 입력 확인
+  const isDataComplete = Object.keys(formData).every(
+    (key) => !formData[key].essential || formData[key].value.trim() !== "",
```

Comment:
[Lv2]
```suggestion
      const isDataComplete = Object.values(formData).every(
    ({ value, essential }: FormItem) => !essential || value.trim() !== "",
  );
```
위처럼 작성하시면 formData 객체에 key로 접근하지않아도 내부 value를 가져올수있습니다!
FormData는 { essential: boolean; value: string } 을 분리한 인터페이스를 지칭하며, 이렇게 하면 인터페이스 재사용으로 타입추론도 가독성좋게 될수있습니다!

Reply:
확인했습니다!